### PR TITLE
Write error information to STDERR

### LINKED ISSUE #4798: Fix CLI printing error information to STDOUT
> Clearly describe the bug

CLI printed error information to STDOUT. This is bad behavior for CLI.

This problem is tricky. If stylelint encounters an error while using the pipeline, the error information may not be output.

> Which rule, if any, is the bug related to?

N/A

> What code is needed to reproduce the bug?

N/A

> What stylelint configuration is needed to reproduce the bug?

N/A

> Which version of stylelint are you using?

`13.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI
```sh
stylelint --print-config /dev/null | jq
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

N/A

> What did you expect to happen?

The CLI is expected to print error information to standard error.

> What actually happened (e.g. what warnings or errors did you get)?

Error information is printed to standard output.

### PR DESCRIPTION
**Pull request by naokikimura**
> Which issue, if any, is this issue related to?

Closes #4798

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Comment by naokikimura** [comment]
Thank you, @malsf21 

Even if there are adverse effects from this change, there is a workaround, so the ripple effect should be small.

Redirecting the standard error output to standard output as follows gives the same behavior as before.

```sh
stylelint "**/*.css" 2>&1
```

It's hard to separate the information once it's mixed, but it's easy to mix the information that's separated.

File: lib/cli.js [hunk 1]
```diff
@@ -534,10 +534,10 @@ module.exports = (argv) => {
  * @returns {void}
  */
 function handleError(err) {
-	console.log(err.stack); // eslint-disable-line no-console
+	process.stderr.write(err.stack);
 	const exitCode = typeof err.code === 'number' ? err.code : 1;
 
-	process.exit(exitCode); // eslint-disable-line no-process-exit
+	process.exitCode = exitCode;
```
Comment (m-allanson) [inline lib/cli.js:540]: Set an exitCode and wait for node to exit when any pending work is complete. This means we can remove another `// eslint-disable-line` comment. The node docs [recommend against using `process.exit()`](https://nodejs.org/api/process.html#process_process_exit_code) where possible.

**Comment by naokikimura** [comment]
Thank you @m-allanson 

I look forward to the quick release of this fix.

**Comment by m-allanson** [comment]
Updated changelog:

> - Fixed: write error information to `stderr` ([#4799](https://github.com/stylelint/stylelint/pull/4799)).

**Pull request merged**